File: DSA/linear search/linearsearch/Linearsearch.java
package linearsearch;

public class Linearsearch{

public static void main(String[] args){


 int [] numbers = {4,6,8,9,0};

 int target = 8;

 int ans = linearsearchfun(numbers, target);

 int ans2 = linearsearchfun2(numbers, target);
 System.out.println(ans);
  System.out.println(ans2);


}

// search in the array: return the index if item found
// otherwise if item not found return -1

static int linearsearchfun(int[] arr,int target) {

  if (arr.length== 0 ){
    
    return -1;

  }

//   run a for loop
for(int index = 0;index < arr.length; index++){

    //  check for element att every index if it is = target

    int element = arr[index];
    
    if(element == target) {

        return index;
    }

 

}


//  this line will execute if none of the return 

// hence the target not found

return Integer.MAX_VALUE;

}


// search the target and return a element
static int linearsearchfun2(int[] arr,int target) {

  if (arr.length== 0 ){
    
    return -1;

  }

//   run a for loop
for(int index = 0;index < arr.length; index++){

    //  check for element att every index if it is = target

    int element = arr[index];
    
    if(element == target) {

        return element;
    }

 

}


//  this line will execute if none of the return 

// hence the target not found

return Integer.MAX_VALUE;

}





// search target and return true 
static boolean linearsearchfun3(int[] arr,int target) {

  if (arr.length== 0 ){
    
    return false;

  }

//   run a for loop
for(int index = 0;index < arr.length; index++){

    //  check for element att every index if it is = target

    int element = arr[index];
    
    if(element == target) {

        return true;
    }

 

}


//  this line will execute if none of the return 

// hence the target not found

return false;



}




}

File: DSA/linear search/linearsearch.cpp
#include <iostream>

// Linear search function


int linearSearch(int arr[], int n, int target) {
    for (int i = 0; i < n; ++i) {
        if (arr[i] == target) {
            return i; 
        }
    }
    return -1;
}

int main() {
    const int size = 10;
    int array[size] = {2, 5, 8, 12, 16, 23, 37, 42, 51, 64};
    int target = 23;

    int result = linearSearch(array, size, target);

    if (result != -1) {
        std::cout << "Element found at index " << result << std::endl;
    } else {
        std::cout << "Element not found in the array" << std::endl;
    }

    return 0;
}
